sign/rpmgensig.c

```diff
@@ -593,8 +634,14 @@ static int rpmSign(const char *rpm, int deleting, int flags)
 	    goto exit;
     }
 
+    if (flags & RPMSIGN_FLAG_FSVERITY) {
+	if (includeVeritySignatures(fd, &sigh, &h))
+	    goto exit;
+    }
+
     if (deleting) {	/* Nuke all the signature tags. */
 	deleteSigs(sigh);
+	deleteFileSigs(sigh);
```

Comment:
I think deleting file signatures needs to be a separate thing from the main package signatures, you might want to delete one but not the other. I guess there's actually no way to remove IMA signatures atm...

Reply:
From my understanding, the package signature covers the file signatures, so we cannot remove them without invalidating the package signature?

If I am misunderstanding this, please let me know.

I also added deletion of the IMA signatures to my patch since I figured they should be deleted.

Reply:
The IMA signatures originally were covered by package signature, but that breaks some fundamental rpm rules so it was changed in a latter release. So these days file signatures are entirely separate items, and can be added and removed without affecting others.

Reply:
> The IMA signatures originally were covered by package signature, but that breaks some fundamental rpm rules so it was changed in a latter release. So these days file signatures are entirely separate items, and can be added and removed without affecting others.

Sweet, I was under the assumption that they were covered, so didn't want to go down that path. I'll have a look at adding this as a separate --delfilesigs option. I think it's reasonable to delete all file signatures with one option, IMA and fsverity, but I can also make it two, if you prefer.

Reply:
Yeah, having a single option for both types of file signatures should be plenty.